cmd/generate/config/rules/azure.go

```diff
@@ -0,0 +1,444 @@
+		RuleID:      "azure-storage-credential-86char",
+		SecretGroup: 1,
+		Regex: generateUniqueTokenRegex(`[ \t]{0,10}[a-zA-Z0-9/+]{86}==`),
+	}
+
+	// validate
+	tps := []string{
+		generateSampleSecret("azure-storage-credential-86char",
+			secrets.NewSecret(alphaNumeric("86") + "==")),
+	}
+	return validate(r, tps, nil)
+}
+
+func AzureStorageCredential43char() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "CSCAN0030, CSCAN0090, CSCAN0150 - Found Azure storage credential in source code file.",
+		RuleID:      "azure-storage-credential-43char",
+		SecretGroup: 1,
+		Regex: generateUniqueTokenRegex(`[a-zA-Z0-9/+]{43}=[^{@\d%]`),
+	}
+
+	// validate
+	tps := []string{
+		generateSampleSecret("azure-storage-credential-43char",
+			secrets.NewSecret(alphaNumeric("43") + "=a")),
+	}
+	return validate(r, tps, nil)
+}
+
+func AzureStorageCredentialSig53() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "CSCAN0030, CSCAN0090, CSCAN0150 - Found Azure storage credential in source code file.",
+		RuleID:      "azure-storage-credential-sig53",
+		SecretGroup: 1,
+		Regex: generateUniqueTokenRegex(`((sig|sas|password)=|>)[a-zA-Z0-9%]{43,53}%3d[^{a-zA-Z0-9%]`),
+	}
+
+	// validate
+	tps := []string{
+		generateSampleSecret("azure-storage-credential-sig53",
+			"sig=" + secrets.NewSecret(alphaNumeric("53") + "%3D")),
+	}
+	return validate(r, tps, nil)
+}
+
+
+func AzureStorageCredentialUserIDPW() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "CSCAN0030 - Found Azure storage credential in source code file.",
+		RuleID:      "azure-storage-credential-useridpw",
+		SecretGroup: 1,
+		Regex: generateUniqueTokenRegex(`((user) ?(id|name)|uid)=.{2,128}?\s*?;\s*?((password)|(pwd))=[^'$%>@'";\[\{][^;"']{2,350}?(;|"|')`),
+	}
+
+	// validate
+	tps := []string{
+		generateSampleSecret("azure-storage-credential-useridpw",
+		"userid=" + secrets.NewSecret(alphaNumeric("128")) + ";password=" + secrets.NewSecret(alphaNumeric("200")) + ";"),
+	}
+	return validate(r, tps, nil)
+}
+
+func AzureStorageCredentialAccountKey() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "CSCAN0030 - Found Azure storage credential in source code file.",
+		RuleID:      "azure-storage-credential-accountkey",
+		SecretGroup: 1,
+		Regex: generateUniqueTokenRegex(`AccountKey\s*=\s*MII[a-zA-Z0-9/+]{43,}?={0,2}`),
+	}
+
+	// validate
+	tps := []string{
+		generateSampleSecret("azure-storage-credential-accountkey",
+			"AccountKey = MII" + secrets.NewSecret(alphaNumeric("43") + "=")),
+	}
+	return validate(r, tps, nil)
+}
+
+func AzureStorageCredentialXStore() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "CSCAN0100 - Found Azure storage credential in source code file.",
+		RuleID:      "azure-storage-credential-xstore",
+		SecretGroup: 1,
+		Regex: generateUniqueTokenRegex(`[<XstoreAccountInfo].*accountSharedKey\s*=\s*['"].*['"]`),
```

Comment:
Just curious, what's the reason for changing some of the patterns?

```diff
-<XstoreAccountInfo[ -~"\s\S\n\r\t]+accountSharedKey\s*=\s*"[^"]{30}[ -~"\s\S\n\r\t]+/>
+[<XstoreAccountInfo].*accountSharedKey\s*=\s*['"].*['"]
```

In particular, this won't work as expected because `[<XstoreAccountInfo]` will only match a single character from inside the square brackets, not `<XstoreAccountInfo`.

Reply:
Probably because there were like 100 rules and I blew through them with all the enthusiasm that I had, now exhausted. When I get some free time I'll try it again, with the above change